metpy/calc/thermo.py

```diff
@@ -840,3 +842,180 @@ def relative_humidity_from_specific_humidity(specific_humidity, temperature, pre
+
+
+@exporter.export
+@check_units('[temperature]', '[pressure]', '[temperature]')
+def get_isentropic_pressure(isentlevs, lev, tmp, *args, **kwargs):
+    r"""Perform isentropic analysis with data in isobaric coordinates.
+
+    Parameters
+    ----------
+    lev : array
+        One-dimensional array of pressure levels
+    tmp : array
+        Array of temperature
+    isentlevs : array
+        One-dimensional array of desired theta surfaces
+
+    Returns
+    -------
+    isentpr : tuple
+        Tuple list with pressure at each isentropic level, followed by each additional
+        argument interpolated to isentopic coordinates.
+
+
+    Other Parameters
+    ----------------
+    args : array, optional
+        Any additional variables will be interpolated to each isentropic level.
+
+    tmpk_out : bool, optional
+        If true, will calculate temperature and output as the last item in the output tuple
+        list. Defaults to False.
+    max_iters : int, optional
+        The maximum number of iterations to use in calculation, defaults to 50.
+    eps : float, optional
+        The desired absolute error in the calculated value, defaults to 1e-6.
+
+    Notes
+    -----
+    Assumes input data in 3-dimensions (no time dimension).Input variable arrays must have the
+    same number of vertical levels as the pressure levels array. Returns calculations for a
+    single timestep. Presure is calculated on isentropic surfaces by assuming that tempertaure
+    varies linearly with the natural log of pressure. Linear interpolation is then used in the
+    vertical to find the pressure at each isentropic level. Interpolation method from
+    [Hoerling1993]_. Any additional arguments are assumed to vary linearlly with ln(p) and will
+    be linearlly interpolated to the new isentropic levels.
+
+    See Also
+    --------
+    potential_temperature
+
+    """
+    # iteration function to be used later
+    def _isen_iter(pk1, isentlevs2, ka, a, b, pok):
+        ekp = np.exp((-ka) * pk1)
+        t = a * pk1 + b
+        f = isentlevs2 - pok * t * ekp
+        fp = pok * ekp * ((ka) * t - a)
+        return pk1 - (f / fp)
+
+    # Raise an error if input data doesn't have 3 dimensions.
+    if tmp.ndim != 3:
+        raise ValueError(str(tmp.ndim) +
+                         ' dimensional data input, 3 dimensional data expected')
+
+    # Change when Python 2.7 no longer supported
+    # Pull out keyword arguments
+    tmpk_out = kwargs.pop('tmpk_out', False)
+    max_iters = kwargs.pop('max_iters', 50)
+    eps = kwargs.pop('eps', 1e-6)
+    # check for pressure in decreasing order, if not, flip.
+    if lev[1] > lev[0]:
+        level = np.flipud(lev) * lev.units
+        tmp = np.flipud(tmp) * tmp.units
+    else:
+        level = lev
+    # convert to appropriate units and check for meteorologically sound data,
+    # if not, raise warning to the user.
+    tmpk = tmp.to('kelvin')
+    levels = level.to('hPa')
```

Comment:
Could probably just do `pressure = pressure.to('hPa')`, there's no need for yet another variable name.